Workflows for compiling and testing

The only GitHub Actions job executing on PRs is a format check. We should add more workflows, at least some that compile and test the extension.

Note: running `npm run compile` in version 0.0.2 shows the following output:
```

> kani-vscode-extension@0.0.2 compile
> tsc -p ./

src/extension.ts:55:48 - error TS2694: Namespace '"vscode"' has no exported member 'StatementCoverage'.

55   const coveredLines = new Map<string, (vscode.StatementCoverage | undefined)[]>();
                                                  ~~~~~~~~~~~~~~~~~

src/extension.ts:93:23 - error TS2339: Property 'StatementCoverage' does not exist on type 'typeof import("vscode")'.

93          ? new vscode.StatementCoverage(0, new vscode.Position(lineNo, 0))
                         ~~~~~~~~~~~~~~~~~

src/extension.ts:133:7 - error TS2339: Property 'coverageProvider' does not exist on type 'TestRun'.

133   run.coverageProvider = {
          ~~~~~~~~~~~~~~~~

src/extension.ts:134:34 - error TS2694: Namespace '"vscode"' has no exported member 'FileCoverage'.

134    provideFileCoverage(): vscode.FileCoverage[] {
                                     ~~~~~~~~~~~~

src/extension.ts:135:28 - error TS2694: Namespace '"vscode"' has no exported member 'FileCoverage'.

135     const coverage: vscode.FileCoverage[] = [];
                               ~~~~~~~~~~~~

src/extension.ts:138:14 - error TS2339: Property 'FileCoverage' does not exist on type 'typeof import("vscode")'.

138       vscode.FileCoverage.fromDetails(
                 ~~~~~~~~~~~~

src/extension.ts:140:43 - error TS2694: Namespace '"vscode"' has no exported member 'StatementCoverage'.

140        statements.filter((s): s is vscode.StatementCoverage => !!s),
                                              ~~~~~~~~~~~~~~~~~


Found 7 errors in the same file, starting at: src/extension.ts:55
```

Maybe this is related to my setup, but if there are additional setup steps we should document them.

--- a/.github/workflows/format-check.yml
+++ b/.github/workflows/format-check.yml
@@ -18,8 +18,13 @@ jobs:
         with:
           node-version: '16.x'
 
-      - name: Check TS code formatting
+      - name: Check TS linting
         run: |
           npm install -g npm@latest
           npm install
           npm run lint
+
+      - name: Check TS formatting
+        run : |
+          npm ci
+          npm run prettier:check:ci

--- a/.prettierignore
+++ b/.prettierignore
@@ -7,3 +7,9 @@ package.lock.json
 out
 .changes
 src/testFixtures/**
+*.json
+.eslintrc.js
+.clang-format
+.prettierrc
+*.d.ts
+*.yml

--- a/package.json
+++ b/package.json
@@ -41,6 +41,7 @@
     "pretest": "npm run compile && npm run lint",
     "lint": "eslint src --ext ts",
     "pretty": "prettier --write \"src/**/*.ts\"",
+    "prettier:check:ci": "./node_modules/.bin/prettier --check .",
     "test": "node ./out/test/runTest.js"
   },
   "devDependencies": {

--- a/src/model/kaniCommandCreate.ts
+++ b/src/model/kaniCommandCreate.ts
@@ -13,10 +13,7 @@ import { createFailedDiffMessage, runKaniCommand } from './kaniRunner';
  * @param args - arguments to Kani if provided
  * @returns verification status (i.e success or failure)
  */
-export async function runKaniHarnessInterface(
-	harnessName: string,
-	args?: number,
-): Promise<any> {
+export async function runKaniHarnessInterface(harnessName: string, args?: number): Promise<any> {
 	let harnessCommand = '';
 	if (args === undefined || isNaN(args)) {
 		harnessCommand = `${KaniConstants.CargoKaniExecutableName} ${KaniArguments.harnessFlag} ${harnessName}`;

--- a/src/model/kaniRunner.ts
+++ b/src/model/kaniRunner.ts
@@ -17,7 +17,6 @@ import { responseParserInterface } from './kaniOutputParser';
  * @returns the path for the binary cargo-kani (either the installed binary or the development one)
  */
 export function getKaniPath(kaniCommand: string): Promise<string> {
-
 	const options = {
 		shell: false,
 	};

--- a/src/test/test-programs/sampleRustString.ts
+++ b/src/test/test-programs/sampleRustString.ts
@@ -203,7 +203,7 @@ export const findHarnessesResultKani = [
 			row: 20,
 			column: 24,
 		},
-		attributes: ['#[kani::unwind(2)]', '#[kani::solver(kissat)]','#[kani::should_panic]'],
+		attributes: ['#[kani::unwind(2)]', '#[kani::solver(kissat)]', '#[kani::should_panic]'],
 		args: {
 			proof: true,
 			test: false,
@@ -394,7 +394,7 @@ export const attributeMetadataUnsupported = [
 			row: 6,
 			column: 13,
 		},
-		attributes: ["#[kani::unwind(0)]", "#[kani::should_panic]", "#[kani::cover]"],
+		attributes: ['#[kani::unwind(0)]', '#[kani::should_panic]', '#[kani::cover]'],
 		args: {
 			proof: true,
 			test: false,
@@ -407,10 +407,10 @@ export const attributeMetadataUnsupported = [
 			row: 14,
 			column: 17,
 		},
-		attributes: ["#[kani::stub]"],
+		attributes: ['#[kani::stub]'],
 		args: {
 			proof: true,
 			test: false,
 		},
-	}
+	},
 ];

--- a/src/ui/concrete-playback/concretePlayback.ts
+++ b/src/ui/concrete-playback/concretePlayback.ts
@@ -54,5 +54,4 @@ function createCommand(
 	} else {
 		return `${KaniConstants.CargoKaniExecutableName} ${KaniArguments.testsFlag} ${KaniArguments.harnessFlag} ${harnessName}`;
 	}
-
 }

--- a/src/ui/sourceCodeParser.ts
+++ b/src/ui/sourceCodeParser.ts
@@ -161,5 +161,4 @@ export namespace SourceCodeParser {
 			}
 		}
 	};
-
 }

